### DISCUSSION TITLE
Update pandas dataframe using relational api

### DISCUSSION THREAD

**Discussion by zachary62** [duckdb/duckdb#4453] [Q&A]
Hey, is there any way to update pandas data frame using relational api?
Currently, we have many examples to query dataframe: https://github.com/duckdb/duckdb/blob/master/examples/python/duckdb-python.py
However, is there any way to update pandas dataframe? For instance, for salary table, can we filter employees with some predicates, then increase their salaries?

**Comment by pdet** [comment]
Hey Zachary, we do not alter dataframes inplace.

An alternative is to create a dataframe from your query output.

e.g.,

```python
import duckdb
import pandas as pd 

con = duckdb.connect()
df = pd.DataFrame.from_dict({"i":[1, 2, 3, 4], "j":["one", "two", "three", "four"]})
# Let's replace df where only with i <=2
df = con.execute("select * from df where i <=2").df()
```

**Reply by zachary62** [reply]
Is this something you consider doing in the future?
I believe this would be an important feature. DuckDB is fast at queries, but updates ("set" clauses) are quite slow, especially for column-wise operators (e.g. increase the salary for all employees).
Therefore, supporting updates for pandas may fill in the gap.

**Reply by pdet** [reply]
I understand that this operation can be expensive, however, supporting updates directly on data frames is not in our current roadmap.
